Client/src/main/java/ch/heigvd/api/calc/Client.java

```diff
@@ -11,29 +14,99 @@
 public class Client {
 
     private static final Logger LOG = Logger.getLogger(Client.class.getName());
+    private final static int PORT = 42000;
+    private final static String IP = "127.0.0.1";
 
     /**
      * Main function to run client
      *
      * @param args no args required
      */
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         // Log output on a single line
         System.setProperty("java.util.logging.SimpleFormatter.format", "%4$s: %5$s%6$s%n");
 
-        BufferedReader stdin = null;
+        final Socket CLIENT_SOCKET;
+        
+        CLIENT_SOCKET = new Socket(IP,PORT);
+        LOG.log(Level.INFO, "Creating a client socket and binding it on any of the available network interfaces and on port {0}",
+                new Object[]{Integer.toString(PORT)});
 
-        /* TODO: Implement the client here, according to your specification
-         *   The client has to do the following:
-         *   - connect to the server
-         *   - initialize the dialog with the server according to your specification
-         *   - In a loop:
-         *     - read the command from the user on stdin (already created)
-         *     - send the command to the server
-         *     - read the response line from the server (using BufferedReader.readLine)
-         */
+        logSocketAddress(CLIENT_SOCKET);
 
-        stdin = new BufferedReader(new InputStreamReader(System.in));
+        // Output du socket pour envoyer les commandes
+        OutputStream output = CLIENT_SOCKET.getOutputStream();
+        PrintWriter writer = new PrintWriter(output, true);
 
+        // Flux pour recevoir les infos du serveur
+        InputStream input = CLIENT_SOCKET.getInputStream();
+        BufferedReader stdin = new BufferedReader(new InputStreamReader(input));
+
+        String msg = "Server info : ";
+        System.out.println(msg);
+        while (!Objects.equals(msg = stdin.readLine(), "end")) {
+            System.out.println(msg);
+        }
+
+        //Compute
+        Scanner sc = new Scanner(System.in);
+        String command = sc.nextLine();
+
+        while(!command.contains("bye")) {
```

Comment:
Je propose plutôt quelque chose comme ça, ça permet d'éviter la répétition de `command = sc.nextline();`  
Le `.trim()` permet de retirer les espaces blancs en début et fin du string.  Ceci pour permettre d'utiliser `.equalsIgnoreCase("bye")` ou `.equals("bye")` plus tôt que le `.contains("bye")` pour ne pas compromettre
une éventuelle commande contenant `bye`. On sait jamais 🤷🏽 

```suggestion
        while(true) {
            // Nouvelle commande de l'utilisateur
            command = sc.nextLine().trim();

            if (command.equalsIgnoreCase("bye")) {
                writer.println("bye");
                break;
            }
```